redux/reducers.js

```diff
@@ -0,0 +1,60 @@
+import { combineReducers } from 'redux';
+import * as types from "./actionTypes";
+
+/* Current shape of state:
+ {
+  user: "David",
+  currentChat: id 
+  chats: [ { id: 1,
+    name: 'Tea Chats',
+    users: ['David', 'Nick'] },
+    { id: 2,
+      name: 'Coffee Chats',
+      users: ['David', 'Ryan'] } ]
+  messages: [
+    { id: 0, chats_id: 1, user: 'David', message: 'ayyyyy', emojis: { David: ':joy:', Nick: ':laughing:' } },
+    { id: 1, chats_id: 1, user: 'Nick', message: 'lmao', emojis: {} }
+  ]
+} */
+
+const messageInitialState = {
+  messages: []
+};
+
+const userInitialState = {
+  user: '',
+  currentChat: 0,
+  chats: []
+};
+
+function messageReducer(state = messageInitialState, action) {  
```

Comment:
we can keep it like this for now since it's small, but convention would be creating separate reducer files per type, so you would have something like messageReducer.js, userReducer.js

Reply:
I think this would be good to change. You can see project structure recommendations in the [Redux documentation](https://redux.js.org/faq/code-structure#what-should-my-file-structure-look-like-how-should-i-group-my-action-creators-and-reducers-in-my-project-where-should-my-selectors-go). 

Looks like what we're following is the "rails-style", is that right, @mnicole? In that case, we'll want separate directories for reducers called `reducers` and actions called `actions`.

Reply:
I'll definitely at least make the separate reducer files change, I was aware of how a lot of Redux projects were structured but really kind of found it easier starting out with it to do one file. Now that it's together in some form they should of course have their own files.

I'm less convinced about making a different directory for each is necessary, I don't think our Redux implementation will ultimately be very complex. This might just be my naivety but very nested file structures seem kind of excessive to me, navigating some of the Redux projects out there it's kind of confusing for them to have tons of ambiguous index.js's under separate actions, reducers, etc. folders, it basically just reduces the accessibility of the code. But yeah I'm open to it, that was just my impression being new to it and browsing some of these Github projects.

Reply:
Ah just saw you all's other comments below on this, I don't mind making different folders actually, I can see how redux/ folder shouldn't be there so might as well replace with actions/ reducers/. I do think the constants folder is useful as well.